core/src/test/java/org/apache/calcite/test/JdbcFrontJdbcBackTest.java

```diff
@@ -73,6 +79,75 @@ class JdbcFrontJdbcBackTest {
         });
   }
 
+
+  /**
+   * Sample subclass used in {@link JdbcFrontJdbcBackTest#testTablesExtraColumn()}.
+   */
+  public static class MetaExtraTable extends CalciteMetaTable {
+    public final String extraLabel;
+    MetaExtraTable(Table calciteTable, String tableCat,
+        String tableSchem, String tableName) {
+      super(calciteTable, tableCat, tableSchem, tableName);
+      this.extraLabel = "extraLabel1";
+    }
+  }
+
+  /** Sample factory that creates MetaExtraTables. */
+  public static class MetaExtraTableFactoryImpl implements CalciteMetaTableFactory {
+
+    public static final MetaExtraTableFactoryImpl INSTANCE = new MetaExtraTableFactoryImpl();
+
+    MetaExtraTableFactoryImpl() {}
+
+    @Override public MetaTable newMetaTable(Table table, String tableCat, String tableSchem,
+        String tableName) {
+      return new MetaExtraTable(table, tableCat, tableSchem, tableName);
+    }
+
+    @Override public String[] getColumnNames() {
+      // 11 columns total.
+      return new String[] {
```

Comment:
Would it be an error if the columns were not prefixed by the list of standard columns? (e.g. adding an extra column in the middle, or removing a column, or re-ordering columns)

Figure out what the necessary constraints are, and add a test that checks that those constraints are being enforced.